File: flex/button/button.c
#include <wiringPi.h>
#include <stdio.h>

//Pin 23 on Raspberry Pi corresponds to BCM GPIO 11 and wiringPi pin 14
#define ButtonPin 14

int main(void)
{
	if(-1 == wiringPiSetup())
	{
			printf("setup wiringPi failed!");
			return 1;
	}

	//Set GPIO pin
	pinMode(ButtonPin, INPUT);
	pullUpDnControl(ButtonPin, PUD_UP);

	printf("Press the button and release it after at least a second.\n");

	//Monitor the button and print text if button
	//has been released after at least a second
	int buf = HIGH;
	while(1)
	{
		int status = digitalRead(ButtonPin);
		if ( (HIGH == status) && (LOW == buf) )
		{
			printf("Button released!\n");
		}
		buf = status;
		delay(1000);
	}

	return 0;
}

File: flex/button/python/button.py
# coding=utf-8
import signal
import sys
import time

import RPi.GPIO as GPIO

# Button - pin 23 on Raspberry Pi corresponds to BCM GPIO 11
ButtonPin = 23


def close(signal, frame):
    GPIO.cleanup()
    sys.exit(0)


# to use Raspberry Pi board pin numbers
GPIO.setmode(GPIO.BOARD)

# set up GPIO input channel
GPIO.setup(ButtonPin, GPIO.IN, GPIO.PUD_UP)

# handle Ctll+C
signal.signal(signal.SIGINT, close)

print('Press the button and release it after at least a second.')

buf = True
while True:
    status = GPIO.input(ButtonPin)
    if status and not buf:
        print('Button released!')
    buf = status
    time.sleep(1)
